dotCMS/src/main/java/com/dotcms/visitor/filter/servlet/VisitorFilter.java

```diff
@@ -107,17 +109,15 @@ private void setVanityAsAttribute(HttpServletRequest request)
             return;
         }
         // Getting the site form the request
-        Host host = WebAPILocator.getHostWebAPI().getCurrentHost(request);
-        final long languageId = this.languageWebAPI.getLanguage(request).getId();
-        if (this.urlUtil.isVanityUrl(uri, host, languageId)) {
-            CachedVanityUrl vanityUrl = APILocator.getVanityUrlAPI().getLiveCachedVanityUrl(
-                    ("/".equals(uri) ? CMS_HOME_PAGE : uri.endsWith("/") ? uri.substring(0, uri.length() - 1) : uri), host,
-                    languageId, this.userWebAPI.getUser(request));
-            
-            request.setAttribute(VANITY_URL_ATTRIBUTE, vanityUrl.getVanityUrlId());
-            
-            
+        final Host host = WebAPILocator.getHostWebAPI().getCurrentHost(request);
+        final Language language = this.languageWebAPI.getLanguage(request);
+
+        Optional<CachedVanityUrl> vanityUrl = APILocator.getVanityUrlAPI().resolveVanityUrl(uri, host, language);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'vanityUrl' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)